extensions/wikia/ImageReview/ImageListGetter.class.php

```diff
@@ -0,0 +1,382 @@
+	 * 2.) Check if there are any outstanding reviews (reviews that this reviewer started
+	 *     but hasn''t finish by making a decision on the images).
+	 * 3.) Create a new review
+	 */
+	public function getImageList() {
+		$this->fetchPreviousReviewFromTimestamp();
+
+		if ( !$this->fetchedEnoughImages() ) {
+			$this->fetchUnfinishedReview();
+		}
+
+		if ( !$this->fetchedEnoughImages() ) {
+			$this->fetchNewReview();
+		}
+
+		return $this->imageList;
+	}
+
+	private function fetchPreviousReviewFromTimestamp() {
+		$this->fetchAndPrepareImageResults( function() {
+			return ( new WikiaSQL() )
+				->SELECT_ALL()
+				->FROM( 'image_review' )
+				->WHERE( 'reviewer_id' )->EQUAL_TO( $this->userId )
+				->AND_( 'review_start' )->EQUAL_TO( $this->timestamp )
+				->ORDER_BY( ImageReviewOrderGetter::PRIORITY_LATEST_SQL )
+				->LIMIT( self::LIMIT_IMAGES_FROM_DB )
+				->runLoop( $this->getDatawareDB(), function ( &$images, $row ) {
+					$images[] = $row;
+				});
+			},
+			false
+		);
+	}
+
+	private function fetchUnfinishedReview() {
+		$this->fetchAndPrepareImageResults( function() {
+			return ( new WikiaSQL() )
+				->SELECT_ALL()
+				->FROM( 'image_review' )
+				->AND_( 'reviewer_id' )->EQUAL_TO( $this->userId )
+				->AND_( 'state' )->EQUAL_TO( $this->getNewState() )
+				->ORDER_BY( ImageReviewOrderGetter::PRIORITY_LATEST_SQL )
+				->LIMIT( self::LIMIT_IMAGES_FROM_DB )
+				->runLoop( $this->getDatawareDB(), function ( &$images, $row ) {
+					$images[] = $row;
+				});
+			},
+			false
+		);
+	}
+
+	private function fetchNewReview() {
+		$this->fetchAndPrepareImageResults( function() {
+			return ( new WikiaSQL() )
+				->SELECT_ALL()
+				->FROM( 'image_review' )
+				->WHERE( 'state' )->EQUAL_TO( $this->state )
+				->AND_( 'top_200' )->EQUAL_TO( 0 )
+				->ORDER_BY( $this->order )
+				->LIMIT( self::LIMIT_IMAGES_FROM_DB )
+				->runLoop( $this->getDatawareDB(), function ( &$images, $row ) {
+					$images[] = $row;
+				});
+			},
+			true
+		);
+	}
+
+	private function fetchAndPrepareImageResults( callable $getImagesQuery, bool $creatingNewReview ) {
+		$deleteFromQueueList = [];
+		$iconsWhere = [];
+
+		foreach ( $getImagesQuery() as $row ) {
+			if ( $this->fetchedEnoughImages() ) {
+				break;
+			}
+
+			// Grab this image now to eliminate any race conditions with other reviewers
+			if ( $creatingNewReview && !$this->assignImageToUser( $row ) ) {
+				// If we failed to assign this image to ourselves, try the next
+				continue;
+			}
+
+			$imageInfo = $this->getImageData( $row );
+
+			// If we failed to load this image, remove it from the review queue
+			if ( !empty( $imageInfo['error'] ) ) {
+				$deleteFromQueueList[] = [
+					'wiki_id' => $row->wiki_id,
+					'page_id' => $row->page_id,
+					'reason' => $imageInfo['error'],
+				];
+				continue;
+			}
+
+			// If this is an icon, put it in a separate queue
+			if ( strpos( 'ico', $imageInfo['extension'] ) ) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "extension" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)